fetch.js

```diff
@@ -195,44 +187,42 @@
       }
     }
 
-    if (support.blob) {
-      this.blob = function() {
-        var rejected = consumed(this)
-        if (rejected) {
-          return rejected
-        }
 
-        if (this._bodyBlob) {
-          return Promise.resolve(this._bodyBlob)
-        } else if (this._bodyFormData) {
-          throw new Error('could not read FormData body as blob')
-        } else {
-          return Promise.resolve(new Blob([this._bodyText]))
-        }
+    this.blob = function() {
+      var rejected = consumed(this)
+      if (rejected) {
+        return rejected
       }
 
-      this.arrayBuffer = function() {
-        return this.blob().then(readBlobAsArrayBuffer)
+      if (this._bodyBlob) {
+        return Promise.resolve(this._bodyBlob)
+      } else if (this._bodyFormData) {
+        throw new Error('could not read FormData body as blob')
+      } else {
+        try {
+          return Promise.resolve(new Blob([this._bodyText]))
+        } catch(e) {
+          throw new Error('Failed to construct a Blob, Include a Blob polyfill')
+        }
       }
+    }
 
-      this.text = function() {
-        var rejected = consumed(this)
-        if (rejected) {
-          return rejected
-        }
+    this.arrayBuffer = function() {
+      return this.blob().then(readBlobAsArrayBuffer)
+    }
 
-        if (this._bodyBlob) {
-          return readBlobAsText(this._bodyBlob)
-        } else if (this._bodyFormData) {
-          throw new Error('could not read FormData body as text')
-        } else {
-          return Promise.resolve(this._bodyText)
-        }
+    this.text = function() {
```

Comment:
I like this unconditional definition of `text()` much better than previous code! 👍